### PR TITLE
Active Storage implementation

### PR DESCRIPTION
**Pull request by Zama15**
## Description

- What changes are being made?
Replaced the image attachment logic for the usage of Active Storage gem
- Why are these changes necessary?
Upgrade the performance of the data base and the application
- How have these changes been tested?
Using PUMA localhost and seeding data to test the application

## List of Changes

1. Added the necessary gem to the GemFile and installing with rails ActiveStorage
2. Migrated active storage files.
3. Modified models and controllers to support the gem
4. Created views and forms necessary to create the posts
5. Modified some views which broke the page with some deprecated methods to the new created with ActiveStorage
6. Styled the views 

## Additional Information

Active Storage already tested
image
Form to edit and create posts
image

## Checklist

- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: app/assets/stylesheets/components/_buttons.scss [hunk 1]
```diff
@@ -16,4 +16,67 @@
...
+
+  &::before {
+    background: standard-color('black');
+    box-shadow: inset $bs-all 0 sum-sizes('lg', 'md', 'sm') $body-bg;
+  }
+
+  &:hover::before {
+    box-shadow: inset $bs-all 0 2px $body-bg;
+  }
 }
```
Comment (juperameza) [inline app/assets/stylesheets/components/_buttons.scss]: New line at the end of the file

File: app/assets/stylesheets/components/_forms.scss [hunk 2]
```diff
@@ -141,4 +141,21 @@ input, select, textarea {
...
+  .post-body {
+    @include padding('na');
+    border: $border standard-color('black');
+    border-radius: $br-md;
+
+    textarea {
+      @include padding('md');
+    }
+  }
 }
```
Comment (juperameza) [inline app/assets/stylesheets/components/_forms.scss]: Same here

File: app/views/posts/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  <div class="post-body">
+    <%= f.text_area :body, size: "70x5", value: @post.body, placeholder: "Write your thoughts here" %>
+  </div>
+  <div class="form-submit">
+    <%= f.submit "Post", class: 'btn btn-submit' %>
+  </div>
+  <div class="post-footer">
+    <p>CATEGORIES</p>
+  </div>
+<% end %>
```
Comment (juperameza) [inline app/views/posts/_form.html.erb]: New line

File: app/views/posts/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<div class="post-container">
+  <%= render 'form' %>
+</div>
```
Comment (juperameza) [inline app/views/posts/edit.html.erb]: New line

File: app/views/posts/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<div class="post-container">
+  <%= render 'form' %>
+</div>
```
Comment (juperameza) [inline app/views/posts/new.html.erb]: New line

**Review by juperameza — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/components/_buttons.scss (hunk 1 above)
Comment (Zama15) [reply app/assets/stylesheets/components/_buttons.scss]: already add the files.insertFinalNewline to true in the vscode :+1:

**Pull request merged**